src/Testcontainers.LowkeyVault/LowkeyVaultBuilder.cs

```diff
@@ -0,0 +1,334 @@
+        /// </summary>
+        /// <param name="input">The input string to parse.</param>
+        /// <returns>A dictionary representation of the key-value pairs in the input string.</returns>
+        static Dictionary<string, string> ParseToDictionary(string input)
+        {
+            var dictionary = new Dictionary<string, string>();
+            var parts = input.Split([' '], StringSplitOptions.RemoveEmptyEntries);
+            string currentKey = null;
+
+            foreach (var part in parts)
+            {
+                if (part.StartsWith("--"))
+                {
+                    var equalIndex = part.IndexOf('=');
+
+                    if (equalIndex > 0)
+                    {
+                        dictionary[part.Substring(0, equalIndex)] = part.Substring(equalIndex + 1);
+                    }
+                    else
+                    {
+                        currentKey = part;
+                    }
+                }
+                else if (currentKey != null)
+                {
+                    dictionary[currentKey] = part;
+                    currentKey = null;
+                }
+            }
+
+            return dictionary;
+        }
+    }
+
+    private void ValidateVaultNames(HashSet<string> vaultNames)
+    {
+        if (!NoAutoRegistration.SetEquals(vaultNames))
+        {
+            var invalid = vaultNames.Where(s => !Regex.IsMatch(s ?? string.Empty, "^[0-9a-zA-Z-]+$", RegexOptions.Compiled)).ToList();
+
+            if (invalid.Count != 0)
+            {
+                throw new ArgumentException("VaultNames contains invalid values: " + string.Join(", ", invalid));
+            }
+        }
+    }
+
+    private static void ValidateAliasMap(Dictionary<string, HashSet<string>> aliasMap)
+    {
+        foreach (var host in aliasMap.Keys)
+        {
+            if (!Regex.IsMatch(host, "^[0-9a-z\\-_.]+$"))
+            {
+                throw new ArgumentException($"Vault host names must match '^[0-9a-z\\-_.]+$'. Found: {host}");
+            }
+        }
+
+        foreach (var alias in aliasMap.Values)
+        {
+            foreach (var host in alias)
+            {
+                if (!Regex.IsMatch(host, "^[0-9a-z\\-_.]+(:[0-9]+|:<port>)?$"))
+                {
+                    throw new ArgumentException($"Vault aliases must match '^[0-9a-z\\-_.]+(:[0-9]+|:<port>)?$'. Found: {host}");
+                }
+            }
+        }
+    }
+
+    private static string ProcessVaultAliases(Dictionary<string, HashSet<string>> aliasMap)
+    {
+        return aliasMap.OrderBy(pair => pair.Key) // Sort keys
+                       .SelectMany(pair => pair.Value.OrderBy(alias => alias) // Sort values
+                       .Select(alias => $"{pair.Key}={alias}"))
+                       .Aggregate((current, next) => $"{current},{next}"); // Join the pairs into a single string with commas
+    }
```

Comment:
Do we really need this? What's wrong with the current approach that overrides the values of existing environment variables? Typically, we don't add much input validation to the module; this is the responsibility of the container. If developers pass invalid arguments to the image, the container should log the issue.

Reply:
Unfortunately, Lowkey Vault has only one environment variable (LOWKEY_ARGS). this environment variable is a concatenation of all the parameters needed for Lowkey Vault to function. that is why we need the `MergeEnv` method.